### ISSUE TITLE
Select Feature inconsistently unresponsive

### ISSUE THREAD

**Issue by Oisin-JK** [opengisch/QField#4751]
Describe the issue
When trying to select an existing feature, QField is inconsistently unresponsive. At the same time it is unresponsive, the cross hairs for plotting new features is offset from screen centre (it is not locked to coordinate position.

I have tried zomming in and out, switching layers, turning edit layer on and off in an attempt to resolve the issue. This does not work. Sometimes it resolves itself. Sometimes I have to exist and restart Qfield to resolve. 
### Reproduction steps
Cannot reprodice reliably.

### Expected behavior
Select feature should be responsive. 

### Observed behavior
See attached 
https://github.com/opengisch/QField/assets/57039094/5fc575a3-bb92-40d5-9c2b-882cda4d48a2

### Desktop (please complete the following information)
 - OS: WIndows 10
 - QGIS Version 3.28.11
 - QFieldSync Version 4.6.0

### Mobile (please complete the following information)
 - Device: Samsung S22 Ultra 5G
 - OS: [Android 13, One UI 5.1
 - QField version: 3.0.6 (d36352) 

### Additional information
* Problem started happening since QField 3.0.1. Didn't happen in an older version of QField: 
* Problem canNOT be reliably reproduced
* Problem happens with all files and projects, not only some files or projects: [Yes]

**Comment by nirvn** [comment]
@Oisin-JK , are you able to consistently trigger this unresponsiveness (ie the mislocated cursor)? I've seen it once or twice but having a hard time replicating

**Comment by BoswachterMarc** [comment]
For your information: We don't experience such problems
**Desktop (please complete the following information)**

    OS: Ubuntu 22.04.3 LTS
    QGIS Version 3.34.0
    QFieldSync Version 4.6.0

**Mobile (please complete the following information)**

    Device: different Android devices
    OS: Android 13, One UI 5.1
    QField version: 3.0.6 (d36352)

**Comment by Oisin-JK** [comment]
> @Oisin-JK , are you able to consistently trigger this unresponsiveness (ie the mislocated cursor)? I've seen it once or twice but having a hard time replicating

No. It is a relatively large project. 2500 points, 300 polygons, related non spatial tables. Rule based styling.

**Comment by nirvn** [comment]
@Oisin-JK , you can try these APKs: https://github.com/opengisch/QField/pull/4755 -- I suspect it'll alleviate your situation.

**Comment by nirvn** [comment]
@Oisin-JK , the fix has been deployed in QField 3.0.7 on play store, eager to hear if that helped you.
